DelvUI/Interface/DragoonHudWindow.cs

```diff
@@ -72,196 +73,102 @@ private void DrawChaosThrustBar()
+            BarBuilder builder = BarBuilder.Create(cursorPos, _config.ChaosThrustBarSize);
+            Bar bar = builder.AddInnerBar(duration, 24f, _config.ChaosThrustBarColor.Map)
+                             .SetBackgroundColor(EmptyColor["background"])
+                             .Build();
 
             if (_config.ShowChaosThrustBarText && duration > 0f)
             {
-                string durationText = duration.ToString();
-                Vector2 textSize = ImGui.CalcTextSize(durationText);
-                DrawOutlinedText(duration.ToString(), new Vector2(cursorPos.X + 5f, cursorPos.Y + _config.ChaosThrustBarSize.Y / 2f - textSize.Y / 2f));
+                builder.SetTextMode(BarTextMode.Single).SetText(BarTextPosition.CenterMiddle, BarTextType.Current);
             }
-
-            drawList.AddRect(cursorPos, cursorPos + _config.ChaosThrustBarSize, 0xFF000000);
+            ImDrawListPtr drawList = ImGui.GetWindowDrawList();
+            bar.Draw(drawList, PluginConfiguration);
         }
 
         private void DrawEyeOfTheDragonBars()
         {
             DRGGauge gauge = PluginInterface.ClientState.JobGauges.Get<DRGGauge>();
 
-            Vector2 cursorPos = Origin
-                              + new Vector2(
-                                    _config.EyeOfTheDragonBarPosition.X - _config.EyeOfTheDragonBarSize.X,
-                                    _config.EyeOfTheDragonBarPosition.Y - _config.EyeOfTheDragonBarSize.Y / 2f
-                                );
+            Vector2 cursorPos = Origin + _config.EyeOfTheDragonBarPosition - _config.EyeOfTheDragonBarSize / 2f;
 
             byte eyeCount = gauge.EyeCount;
-            ImDrawListPtr drawList = ImGui.GetWindowDrawList();
-
-            for (byte i = 0; i < 2; i++)
-            {
-                Vector2 size = _config.EyeOfTheDragonBarSize;
-                size.X = size.X - _config.EyeOfTheDragonBarPadding / 2f;
 
-                cursorPos = new Vector2(cursorPos.X + (size.X + _config.EyeOfTheDragonBarPadding * i) * i, cursorPos.Y);
-
-                if (eyeCount >= i + 1)
-                {
-                    drawList.AddRectFilledMultiColor(
-                        cursorPos,
-                        cursorPos + size,
-                        _config.EyeOfTheDragonColor.Map["gradientLeft"],
-                        _config.EyeOfTheDragonColor.Map["gradientRight"],
-                        _config.EyeOfTheDragonColor.Map["gradientRight"],
-                        _config.EyeOfTheDragonColor.Map["gradientLeft"]
-                    );
-                }
-                else
-                {
-                    drawList.AddRectFilled(cursorPos, cursorPos + size, EmptyColor["background"]);
-                }
-
-                drawList.AddRect(cursorPos, cursorPos + size, 0xFF000000);
-            }
+            BarBuilder builder = BarBuilder.Create(cursorPos, _config.EyeOfTheDragonBarSize);
+            Bar eyeBars = builder.SetChunks(2)
```

Comment:
Default value for EyeOfTheDragonbarSize.X needs to be doubled when migrating from individual chunk drawing to BarBuilder (as it uses total bar width and not chunk width).